frontend/src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,61 @@
+import { ReactNode, useCallback, useEffect } from "react";
+import { createPortal } from "react-dom";
+import styles from "./Modal.module.scss";
+import ExitIcon from "../../assets/ExitIcon.svg";
+
+export interface ModalProps {
+    title?: string;
+    description?: string;
+    children?: ReactNode;
+    visible: boolean;
+    onClose: () => void;
+    outerClassName?: string;
+    containerClassName?: string;
+}
+
+const Modal = ({ title, visible, onClose, children, description = "" }: ModalProps) => {
+    const escFunction = useCallback(
+        (e) => {
+            if (e.keyCode === 27) {
```

Comment:
suggestion: use `event.key` instead of `keyCode` which is deprecated. You'll most likely have to type the `event` object as `KeyboardEvent`.

https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key